hedera-mirror-common/src/main/java/com/hedera/mirror/common/domain/transaction/RoyaltyFee.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.common.domain.transaction;
+
+import com.hedera.mirror.common.domain.entity.EntityId;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@AllArgsConstructor(access = AccessLevel.PRIVATE) // for Builder
+@Builder
+@Data
+@NoArgsConstructor
+public class RoyaltyFee {
+
+    private Boolean allCollectorsAreExempt;
+
+    private EntityId collectorAccountId;
+
+    private FixedFee fallbackFee;
+
+    private Long royaltyDenominator;
+
+    private Long royaltyNumerator;
```

Comment:
Can name similarly then pull amount to parent.

```suggestion
    private long amountDenominator;

    private long amount;
```

Reply:
RoyaltyFee has a FallbackFee property, which itself includes an `amount`, so having `amount` within RoyaltyFee and within FallbackFee seemed confusing, I have left it as `royaltyNumerator` thus far. 

Not all three of the fee types included the `amountDenominator`, so the `AbstractFee` does not include it, and the `royaltyDenominator` remains in the RoyaltyFee class.

Reply:
Thanks I have made these changes, also `amountDenominator` was renamed `denominator` per Xin's feedback.